### DISCUSSION TITLE
Memory leak while Compiling

### DISCUSSION THREAD

**Discussion by AndreiTC** [adoconnection/RazorEngineCore#103] [General]
We have a report containing 10-20 dynamic pdf pages which change very often. 
Every time we call RazorEngine.Compile there are 25x*(10 or 20, number of pages)MB of memory which is not deallocated.
Are you aware of such issues? And if how can we mitigate it?
![image](https://user-images.githubusercontent.com/32672865/161605564-bf2531de-f171-40b6-be2a-7e58dc05c4f7.png)

What we basically do is :

private IRazorEngineCompiledTemplate<TemplateBase<TTemplateViewModel>> CompileTemplate<TTemplateViewModel>(Type baseViewModelType, string templateContent)
        {
            return engine.Compile<TemplateBase<TTemplateViewModel>>(templateContent, ConfigureEngine);

            void ConfigureEngine(IRazorEngineCompilationOptionsBuilder b)
            {
                b.AddAssemblyReference(baseViewModelType);
                b.AddAssemblyReference(typeof(ColorTranslator));
        
                TryAddAssemblyReferenceByName(b, "System.Collections, Version=4.1.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a");
                TryAddAssemblyReferenceByName(b, "System.Resources.ResourceManager, Version=4.1.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a");
            }

            static void TryAddAssemblyReferenceByName(IRazorEngineCompilationOptionsBuilder b, string assemblyName)
            {
                try
                {
                    b.AddAssemblyReferenceByName(assemblyName);
                }
                catch
                {
                    // if the assembly fails to load that's fine, it just means that the
                    // app didn't depend on it (e.g. because it was still .NET 4.7.2)
                }
            }
        }

**Comment by adoconnection** [comment]
Hi, have a look at latest gitter conversation: https://gitter.im/RazorEngineCore/community?utm_source=badge&utm_medium=badge&utm_content=badge

Can you reproduce this problem in console app?
In short, you are not supposed to compile templates over and over, and rather use caching.
